From 0057b9fa1576d1222be1b078181248c8281c4f23 Mon Sep 17 00:00:00 2001
From: Ross Wille <wille@freescale.com>
Date: Wed, 19 Nov 2008 19:17:03 -0700
Subject: [PATCH] ENGR00099614-2 MX25: Enable MC34704 PMIC

Enable everything necessary to use the MC34704
PMIC chip on the MX25 3stack board.

Signed-off-by: Ross Wille <wille@freescale.com>
---
 arch/arm/configs/imx25_3stack_defconfig |   24 ++++++++++++++++++++++--
 1 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx25_3stack_defconfig b/arch/arm/configs/imx25_3stack_defconfig
index d6aadda..0b44cf6 100644
--- a/arch/arm/configs/imx25_3stack_defconfig
+++ b/arch/arm/configs/imx25_3stack_defconfig
@@ -175,6 +175,8 @@ CONFIG_ARCH_MXC=y
 # CONFIG_ARCH_MX27 is not set
 CONFIG_ARCH_MX25=y
 # CONFIG_ARCH_MX21 is not set
+CONFIG_I2C_MXC_SELECT1=y
+# CONFIG_I2C_MXC_SELECT2 is not set
 
 #
 # MX25 Options
@@ -189,6 +191,7 @@ CONFIG_ARCH_MXC_HAS_NFC_V2_1=y
 #
 # Device options
 #
+# CONFIG_I2C_MXC_SELECT3 is not set
 CONFIG_DMA_ZONE_SIZE=24
 
 #
@@ -528,7 +531,11 @@ CONFIG_MTD_NAND_MXC_V2=y
 #
 # Voltage and Current regulators
 #
-# CONFIG_REGULATOR is not set
+CONFIG_REGULATOR_API=y
+CONFIG_REGULATOR=y
+# CONFIG_REGULATOR_DEBUG is not set
+CONFIG_REGULATOR_MC34704=y
+# CONFIG_REGULATOR_WM8350 is not set
 # CONFIG_PARPORT is not set
 CONFIG_BLK_DEV=y
 # CONFIG_BLK_DEV_COW_COMMON is not set
@@ -718,7 +725,15 @@ CONFIG_HW_RANDOM=y
 # CONFIG_R3964 is not set
 # CONFIG_RAW_DRIVER is not set
 # CONFIG_TCG_TPM is not set
-# CONFIG_I2C is not set
+CONFIG_I2C=y
+CONFIG_I2C_BOARDINFO=y
+# CONFIG_I2C_CHARDEV is not set
+
+#
+# I2C Hardware Bus support
+#
+CONFIG_I2C_MXC=y
+# CONFIG_I2C_MXC_HS is not set
 # CONFIG_I2C_SLAVE is not set
 CONFIG_SPI=y
 CONFIG_SPI_MASTER=y
@@ -997,7 +1012,12 @@ CONFIG_RTC_INTF_DEV=y
 #
 # MXC PMIC support
 #
+CONFIG_MXC_PMIC=y
 # CONFIG_MXC_PMIC_MC13783 is not set
+# CONFIG_MXC_PMIC_MC13892 is not set
+CONFIG_MXC_PMIC_MC34704=y
+# CONFIG_MXC_PMIC_MC9SDZ60 is not set
+CONFIG_MXC_PMIC_CHARDEV=y
 
 #
 # Advanced Power Management devices
-- 
1.5.4.4

